MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -4847,37 +4842,38 @@ public TargetRoll getTargetFor(IPartWork partWork, Person tech) {
             target.addModifier(getFaction().getEraMod(getGameYear()), "era");
         }
 
-        boolean isOvertime = false;
+        final boolean isOvertime;
         if (isOvertimeAllowed()
                 && (tech.isTaskOvertime(partWork) || partWork.hasWorkedOvertime())) {
             target.addModifier(3, "overtime");
             isOvertime = true;
+        } else {
+            isOvertime = false;
         }
 
-        int minutes = Math.min(partWork.getTimeLeft(), tech.getMinutesLeft());
-        if (isOvertimeAllowed()) {
-            minutes = Math.min(minutes, tech.getMinutesLeft() + tech.getOvertimeLeft());
+        final int minutes = Math.min(partWork.getTimeLeft(), techTime);
+        if (minutes <= 0) {
+            MekHQ.getLogger().error("Attempting to get the target number for a part with zero time left.");
+            return new TargetRoll(TargetRoll.AUTOMATIC_SUCCESS, "No part repair time remaining.");
         }
+
         int helpMod;
-        if (null != partWork.getUnit() && partWork.getUnit().isSelfCrewed()) {
-            int hits;
-            if (null != partWork.getUnit().getEntity().getCrew()) {
-                hits = partWork.getUnit().getEntity().getCrew().getHits();
-            } else {
-                hits = 6;
-            }
-            helpMod = getShorthandedModForCrews(hits);
+        if ((partWork.getUnit() != null) && partWork.getUnit().isSelfCrewed()) {
+            helpMod = getShorthandedModForCrews((partWork.getUnit().getEntity().getCrew() == null)
```

Comment:
This looks familiar, didn't I see the same logic in the other pull request I just reviewed?

Maybe a better solution would be to unify that logic somewhere.

Reply:
You did, and I've got a change that standardizes the logic in that PR.